test/integration/IntegrationBaseSetup.t.sol

```diff
@@ -71,9 +70,10 @@ abstract contract IntegrationBaseSetup is TestUtils {
 
         _registrar.setVault(_vault);
 
+        // NOTE: Using `DeployBase` as a contract instead of a script, means that the deployer is `_deploy` itself.
         (address minterGateway_, address minterRateModel_, address earnerRateModel_) = _deploy.deploy(
-            _deployer,
-            0,
+            address(_deploy),
+            1,
```

Comment:
Why is the nonce now `1`?

Reply:
Nonce 0 for a contact itself it used when it is deployed. So, the first contract a contract creates uses nonce 1.